Dockerfile

```diff
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1
+
+FROM maven:3.9.8-eclipse-temurin-11-focal as package
+
+WORKDIR /build
+
+COPY ./ ./
+RUN --mount=type=cache,target=/root/.m2 \
+    MAVEN_OPTS=-Dorg.slf4j.simpleLogger.defaultLogLevel=warn mvn  -B  package -DskipTests 
+RUN mv xtable-utilities/target/xtable-utilities-$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)-bundled.jar target/app.jar
+
+FROM eclipse-temurin:17-jre-jammy AS final
+
+# Create a non-privileged user that the app will run under.
+# See https://docs.docker.com/go/dockerfile-user-best-practices/
+ARG UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    appuser
+USER appuser
+
+# Copy the executable from the "package" stage.
+COPY --from=package build/target/app.jar ./app.jar
+
+ENTRYPOINT [ 
+            "java", \
+            "--add-opens=java.base/sun.nio.hb=ALL-UNNAMED", \
```

Comment:
can you add a comment explain why we want these flags?

Reply:
It's necessary to added this flag to run spark on java >= 17